### PR TITLE
Fix all dead qhelp links

### PR DESCRIPTION
**Pull request by smowton**
For those documents with no obvious new home I've pointed the links to the Internet Archive.

### CODE REVIEW

**Review by erik-krogh — COMMENTED** [review]
Lots of the old links aren't dead, they just redirect to another URL. (E.g. all the `ecma-262` links). 

And I don't see how the link in `UnreachableMethodOverloads.qhelp` is dead. 
Both the old and new link works fine, so the change is OK with me. I just don't see why it is needed. 

And the link in `javascript/ql/src/Security/CWE-829/InsecureDownload.qhelp` is changed from OWASP to Wikipedia. 
The man in the middle article from Owasp still exist, it's just been renamed to [`Manipulator-in-the-middle attack`](https://owasp.org/www-community/attacks/Manipulator-in-the-middle_attack).  
I think the Wiki article is better, so it's fine.

**Review by RasmusWL — APPROVED** [review]
:+1: For Python things, thanks :bow:

**Review by MathiasVP — APPROVED** [review]
C/C++ 👍

**Review by hvitved — COMMENTED** [review]
Thanks a lot.

File: csharp/ql/src/Likely Bugs/ReferenceEqualsOnValueTypes.qhelp [hunk 1]
```diff
@@ -25,7 +25,7 @@ really be compared using <code>i == j</code>.</p>
 <references>
 
   <li>MSDN: <a href="http://msdn.microsoft.com/en-us/library/system.object.referenceequals.aspx">Object.ReferenceEquals Method</a>.</li>
-  <li>The Way I See It: <a href="http://blogs.msdn.com/b/vijaysk/archive/2008/03/19/object-referenceequals-valuevar-valuevar-will-always-return-false.aspx">Object.ReferenceEquals(ValueVar, ValueVar) will always return false.</a></li>
+  <li>The Way I See It: <a href="https://web.archive.org/web/20190919103548/https://blogs.msdn.microsoft.com/vijaysk/2008/03/18/object-referenceequalsvaluevar-valuevar-will-always-return-false/">Object.ReferenceEquals(ValueVar, ValueVar) will always return false.</a></li>
```
Comment (smowton) [inline csharp/ql/src/Likely Bugs/ReferenceEqualsOnValueTypes.qhelp:28]:
```suggestion
  <li>The Way I See It: <a href="https://docs.microsoft.com/en-us/archive/blogs/vijaysk/object-referenceequalsvaluevar-valuevar-will-always-return-false">Object.ReferenceEquals(ValueVar, ValueVar) will always return false.</a></li>
```

**Comment by smowton** [comment]
@erik-krogh don't know what happened with ECMA; on Friday a `curl -L -f` was failing re: those links but no longer. Guess I might as well keep the new version since that appears to be the current best reference.

The Typescript spec link in `UnreachableMethodOverloads.qhelp` was indeed alive; I just made it uniform with the others, which I guess might as well point to the latest version as of the time of writing.

**Review by erik-krogh — APPROVED** [review]
JS :+1:

**Review by hvitved — APPROVED** [review]
C# 👍

**Review by MathiasVP — APPROVED** [review]

**Review by aschackmull — APPROVED** [review]

**Pull request merged**